apps/omg_watcher_info/lib/omg_watcher_info/db/block.ex

```diff
@@ -92,6 +92,56 @@ defmodule OMG.WatcherInfo.DB.Block do
     |> Paginator.set_data(paginator)
   end
 
+  @spec query_last_24_hour(Ecto.Query.t()) :: Ecto.Query.t()
+  def query_last_24_hour(query) do
+    now = DateTime.to_unix(DateTime.utc_now())
+    twenty_four_hours = 86_400
+    from(block in query, where: block.timestamp >= ^(now - twenty_four_hours))
+  end
+
+  @doc """
+  Returns the total number of blocks in the last 24 hours
+  """
+  @spec get_count_last_24_hour() :: non_neg_integer()
+  def get_count_last_24_hour do
+    query_count()
+    |> query_last_24_hour()
+    |> DB.Repo.all()
+    |> Enum.at(0)
+  end
+
+  @doc """
+  Returns the total number of blocks
+  """
+  @spec get_count() :: non_neg_integer()
+  def get_count do
+    query_count()
+    |> DB.Repo.all()
+    |> Enum.at(0)
+  end
+
+  @spec query_count :: Ecto.Query.t()
+  defp query_count do
+    from(block in __MODULE__, select: count())
+  end
+
+  @spec query_timestamps :: Ecto.Query.t()
+  defp query_timestamps do
+    from(block in __MODULE__, select: %{timestamp: block.timestamp})
+  end
+
+  @spec get_timestamps :: []
+  def get_timestamps() do
```

Comment:
```suggestion
  def all_timestamps() do
```

Avoid mental load by following [Ecto convention](https://hexdocs.pm/ecto/Ecto.Repo.html) where `all` returns multiple results and `get` returns single.

Reply:
Done 8946c62.